### ISSUE TITLE
Unable to upload HEIF image as money request

### ISSUE THREAD

**Issue by m-natarajan** [Expensify/App#47078]
If you haven’t already, check out our [contributing guidelines](https://github.com/Expensify/ReactNativeChat/blob/main/contributingGuides/CONTRIBUTING.md) for onboarding and email contributors@expensify.com to request to join our Slack channel!
___

**Version Number:** 9.0.18-1
**Reproducible in staging?:** Y
**Reproducible in production?:** N
**If this was caught during regression testing, add the test name, ID and link from TestRail:**
**Email or phone of affected tester (no customers):**
**Logs:** https://stackoverflow.com/c/expensify/questions/4856
**Expensify/Expensify Issue URL:**
**Issue reported by:** @Beamanator 
**Slack conversation:** https://expensify.slack.com/archives/C049HHMV9SM/p1723093506634689

## Action Performed:
1. Take picture on iPhone 12 (HEIF image)
2. Try to request money & upload by scanning receipt
## Expected Result:
 Image should upload & scan
## Actual Result:
Saw error message "Invalid file type"
## Workaround:
Unknown
## Platforms:

Which of our officially supported platforms is this issue occurring on?
- [ ] Android: Native
- [ ] Android: mWeb Chrome
- [x] iOS: Native
- [ ] iOS: mWeb Safari
- [ ] MacOS: Chrome / Safari
- [ ] MacOS: Desktop

## Screenshots/Videos

Add any screenshot/video evidence

https://github.com/user-attachments/assets/5b71a34d-944c-4b17-bc36-ac2548d7238b

<img src="https://github.com/user-attachments/assets/cf852444-e66e-436f-b454-a282d68a4b77" width="40%" height="40%">

</details>

[View all open jobs on GitHub](https://github.com/Expensify/App/issues?q=is%3Aopen+is%3Aissue+label%3A%22Help+Wanted%22)

**Comment by Beamanator** [comment]
Being discussed a bit in slack - https://expensify.slack.com/archives/C01GTK53T8Q/p1723093823149789

**Comment by Beamanator** [comment]
Working well in staging! marking NAB (probably shouldn't close though, since i think we still can't upload HEIF on web)

**Comment by Beamanator** [comment]
@johnmlee101 feel free to assign to me if you want, but totes up to you

**Comment by Beamanator** [comment]
Ok so HEIF image uploads should be working well on iOS now (i use this every day) but I believe it doesn't work in web... @BrtqKr since you worked on https://github.com/Expensify/App/pull/47139 and https://github.com/Expensify/App/pull/46025, are you interested in giving that a shot at a fix?

**Comment by Beamanator** [comment]
OK ya the CP worked but we're keeping this open to hopefully fix HEIF uploads on ALL platforms

**Comment by kevinksullivan** [comment]
No payment necessary. Can we close this out @Beamanator ?

**Comment by Beamanator** [comment]
Can't close b/c we still can't upload HEIF images on web - @BrtqKr are you interested in taking that on? If not, we can open this up to be `External`

**Comment by Beamanator** [comment]
Bump @nyomanjyotisa 🙏

**Comment by johnmlee101** [comment]
Removing overdue since we're waiting ^

**Comment by kevinksullivan** [comment]
Looping in another BZ member as I'm going OOO

**Comment by RachCHopkins** [comment]
@parasharrajat looks like Jack replied to you in Slack.

**Comment by Beamanator** [comment]
I just posted in slack ([here](https://expensify.slack.com/archives/C01GTK53T8Q/p1727812153316009?thread_ts=1707130645.906709&cid=C01GTK53T8Q)) to offer 1 potential alternate solution before we go through the new lib process again, just because people already thought "this was an edge case" so it could be nice to do a small improvement instead of implement the full feature w/ this lib

**Comment by RachCHopkins** [comment]
@Beamanator was there any resolution there?

**Comment by Beamanator** [comment]
Nobody commented 😅  :forever-alone: so I bumped the thread 😬 🙏

**Comment by RachCHopkins** [comment]
Followed up in slack thread.

**Comment by RachCHopkins** [comment]
Followed up again.

**Comment by RachCHopkins** [comment]
hey @Beamanator where are we at with this? I don't see any updates in Slack for a month now.

**Comment by Beamanator** [comment]
yeaaaaa so sorry that's my bad, i'm going to start up that slack convo today 🙏

**Comment by Beamanator** [comment]
@nyomanjyotisa @parasharrajat can y'all please take a look at https://github.com/Expensify/App/issues/52487 before I post in slack, suggesting we use `heic2any`? I would like some help looking into:
1. Are any of the "alternative packages" really viable / usable in our project?
2. Do y'all know if the package would add a bundle size of 2.72 MB (the "unpacked size", mentioned in https://www.npmjs.com/package/heic2any) or less?
3. Any other details before I bring this up for the app deployers to discuss again?

**Comment by johnmlee101** [comment]
Just FYI Apple's Genmoji are sent as HEIC so if we want users to be able to use that, this requires it 😄

**Comment by johnmlee101** [comment]
> If so, could you take a small video showing how it doesn't work with our app so I can put it in the slack thread?

It just uploads it as a HEIF so it looks like a file, not a thumbnail or image.

**Comment by Beamanator** [comment]
Aah that's not great 😅  Luckily we got a lot of 👍 [in slack](https://expensify.slack.com/archives/C01GTK53T8Q/p1731595198635519) to move forward with the new lib! 👍

**Comment by Beamanator** [comment]
Yes sorry, I thought that was clear 😅  We got approval so let's move forward 👍

**Comment by Beamanator** [comment]
Oof yep, seems in https://github.com/alexcorvi/heic2any/issues/61 a lot of people are struggling with that same issue... Thanks for checking if `heic-to` will work well on all platforms @nyomanjyotisa 🙏

**Comment by Beamanator** [comment]
I think @parasharrajat you're right that this bug only happens on web platforms at the moment

**Comment by Beamanator** [comment]
> However, if a receipt is uploaded using the 'Choose File' option, the image is not converted causing the upload to fail.

What exactly fails with that upload? Like right now, uploading HEIC from iOS native works... Right? I assume we convert to jpg in the backend? Why doesn't that work with the new plan?

**Comment by Beamanator** [comment]
Ok ok sorry for the delay here... @nyomanjyotisa since you're assigned here, can you please summarize where your latest proposal stands? If you & @parasharrajat agree we will need a new lib (https://github.com/hoppergee/heic-to) & it looks safe to use, I will have to bring it to slack again & get it approved before we can move forward

**Comment by Beamanator** [comment]
juuuuust got back from vacay, will get to this soon but probably not today

**Comment by Beamanator** [comment]
Sorryyyy - deployer this week so will be pretty busy 😢  will try to get back to this soon

**Comment by Beamanator** [comment]
Ok created the new lib request issue: https://github.com/Expensify/App/issues/55777

I will post in slack once we have a few answers:
1. `What will the effect be on the bundle size of our code?`
2. What will be the process of using this lib in our code? Do we need to build anything on our own devices before running it locally?

@nyomanjyotisa @parasharrajat please take a look and let me know if anything's missing.

**Comment by Beamanator** [comment]
Bump @nyomanjyotisa @parasharrajat 🙏

**Comment by Beamanator** [comment]
Thanks @parasharrajat ! 🙏 

I will bring this to slack next week, as i'm going OOO for pretty much the rest of this week 👍

**Comment by RachCHopkins** [comment]
Hey @Beamanator are you back yet?

**Comment by Beamanator** [comment]
Ooh yeah was a tough week last week, merp! Coming back to this this week! 🙏

**Comment by Beamanator** [comment]
back from my last OOO for a while! Planning to jump back in today

**Comment by Beamanator** [comment]
Posted review link - https://expensify.slack.com/archives/C01GTK53T8Q/p1744200665946799

**Comment by Beamanator** [comment]
bumped the thread

**Comment by Beamanator** [comment]
back from OOO - seems like we're good to move forward!

**Comment by Beamanator** [comment]
@nyomanjyotisa can you please confirm if you've started / continued working on the PR for this? 🙏  just checking b/c more people seem to be hitting this issue

**Comment by parasharrajat** [comment]
Can we increase the price for this job as we have to do a lot of back and forth to get this done? 

1. Long discussions. 
2. Resigning again and again. 
3. Tested and implemented two different implementations based on two libraries. 
4. Being patient with the longer library approval process. We started working on this on [Sep 9, 2024](https://github.com/Expensify/App/issues/47078#issuecomment-2338267308)


The price should be at least $500 for this job. cc: @Beamanator @RachCHopkins

**Comment by Beamanator** [comment]
$500 sounds fair to me 👍

**Comment by parasharrajat** [comment]
# BugZero Checklist:

- [x] **[Contributor]** Classify the bug:

<details>
<summary>Bug classification</summary>


Source of bug:
  - [ ] 1a. Result of the original design (eg. a case wasn't considered)
  - [ ] 1b. Mistake during implementation
  - [ ] 1c. Backend bug
  - [x] 1z. Other:

Where bug was reported:
  - [x] 2a. Reported on production (eg. bug slipped through the normal regression and PR testing process on staging)
  - [ ] 2b. Reported on staging (eg. found during regression or PR testing)
  - [ ] 2d. Reported on a PR
  - [ ] 2z. Other:

Who reported the bug:
  - [ ] 3a. Expensify user
  - [x] 3b. Expensify employee
  - [ ] 3c. Contributor
  - [ ] 3d. QA
  - [ ] 3z. Other:

</details>

- [x] **[Contributor]** The offending PR has been commented on, pointing out the bug it caused and why, so the author and reviewers can learn from the mistake.

    Link to comment: Platform issue.

- [x] **[Contributor]** If the regression was CRITICAL (e.g. interrupts a core flow) A discussion in [#expensify-open-source](https://app.slack.com/client/E047TPA624F/C01GTK53T8Q) has been started about whether any other steps should be taken (e.g. updating the PR review checklist) in order to catch this type of bug sooner.

    Link to discussion: **Not needed**

- [x] **[Contributor]** If it was decided to create a regression test for the bug, please propose the [regression test](https://github.com/Expensify/App/blob/main/contributingGuides/REGRESSION_TEST_BEST_PRACTICES.md) steps using the template below to ensure the same bug will not reach production again.

- [ ] **[BugZero Assignee]** Create a GH issue for creating/updating the regression test once above steps have been agreed upon.

    Link to issue:

## Regression Test Proposal


### Test:


1. Open Create request flow.
2. Add the Receipt image of type heic or heif. 
3. Verify that the image uploads successfully

Do we agree 👍 or 👎

**Comment by RachCHopkins** [comment]
OMG NO WAY! I thought I would never see the day that I could nearly close this issue. I'm so excited. Will pay it out tomorrow!

**Comment by RachCHopkins** [comment]
Payment Summary:

- Contributor: @nyomanjyotisa to be paid $500 via Upwork
- Contributor+: @parasharrajat to be paid $500 via newDot Manual Request

Upwork job [here](https://www.upwork.com/jobs/~0116d43a98c70752e7)

**Comment by RachCHopkins** [comment]
Contributor has been paid, the contract has been completed, and the Upwork post has been closed. 

@parasharrajat you have the 🟢 to request via NewDot.
